src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeActions
+{
+    internal static class CodeActionHelpers
+    {
+        public static async Task<IEnumerable<CodeAction>> GetCodeActionsAsync(
+            Document? document,
+            ICodeFixService codeFixService,
+            ICodeRefactoringService codeRefactoringService,
+            LSP.Range selection,
+            CancellationToken cancellationToken)
+        {
+            if (document == null)
+            {
+                return ImmutableArray<CodeAction>.Empty;
+            }
+
+            var (codeFixCollections, codeRefactorings) = await GetCodeFixesAndRefactoringsAsync(
+                document, codeFixService,
+                codeRefactoringService, selection,
+                cancellationToken).ConfigureAwait(false);
+
+            var codeActions = codeFixCollections.SelectMany(c => c.Fixes.Select(f => f.Action)).Concat(
+                    codeRefactorings.SelectMany(r => r.CodeActions.Select(ca => ca.action)));
```

Comment:
is htis the ordering the host will show the items in?  if so, this seems to not follow the ordering we have in VS (where we sort/order based on priority and other factors).  Is that intentional?  Is that problematic?  For example, rename is a refactoring, and we expect it to be high pri and at the top of the list.  this looks liek this wouldn't support that.

Reply:
we may need to extract out hte logic in SuggestedActionSoruce to fetch and order the actions properly and then use that from that location as well as this one.

Reply:
Opened #45726 to track this.